build_runner_core/lib/src/package_graph/apply_builders.dart

```diff
@@ -175,9 +175,19 @@ class BuilderApplication {
               optionsWithDefaults.overrideWith(BuilderOptions.forRoot);
         }
 
-        var builder = _scopeLogSync(
-            () => builderFactory(optionsWithDefaults), new Logger(builderKey));
-        if (builder == null) throw 'builderFactory did not return a builder.';
+        final logger = new Logger(builderKey);
+        Builder builder;
+        try {
+          builder =
+              _scopeLogSync(() => builderFactory(optionsWithDefaults), logger);
+        } catch (e, st) {
+          logger.severe('Failed to instantiate builder.', e, st);
```

Comment:
nit: log the package and options

Reply:
also use `new JsonEncoder.withIndent('  ').convert(e.options.config)` for the options ;)

Reply:
also note errors won't be thrown here its always just the null case and we already log the error with severe

Reply:
Done